### ISSUE TITLE
Document SDLs in more detail

### ISSUE THREAD

**Issue by Attagoat** [H-uru/Plasma#1454]
Throughout my time playing H'uru I've noticed that there a lot of SDL files and many of the variables are not well documented for what changes what in an age. Will be adding a checklist for myself later on what needs to be documented.

**Comment by Deledrius** [comment]
I suspect that most documentation on which Age SDL variables change which features was internal-only at Cyan, if they had documentation on it at all.  The rest is as-is in the files themselves, or by looking into the Age itself.  Generally the names tend to be at least somewhat self-documenting, but there are more than a few obscure ones.

**Comment by Attagoat** [comment]
@Deledrius I agree with you which is why I am working on adding comments to the SDLs

**Comment by Attagoat** [comment]
Alright, I added the checklist.

**Comment by Hazado** [comment]
RestorationGuild.sdl - This is the SDL file from the CC version of Great Tree Pub.
It wont be used again and it will probably only be used for references if the puzzle in the GTP is ever revamped.

**Comment by dpogue** [comment]
> working on adding comments to the SDLs

Something to be aware of with SDL files is that every new change (adding or removing variables) requires copying and pasting the entire block of SDL declarations as a new version. If you're including inline comments in the file, those will probably also get copied/pasted each time.

That potentially becomes a performance consideration, because right now the SDL files on MOULa are downloaded in full on every launch of the game, and I don't believe there is any file compression (SDL files would compress wonderfully with GZip since they're a lot of duplicated text, but I don't think that's used here).

**Comment by Attagoat** [comment]
Possible solution to this, Could we have an internal SDL with the documentation just for us that doesn't get published to the actual game?

**Comment by dgelessus** [comment]
Perhaps it would make sense to document everything in a single block at the top of each .sdl file, rather than inside the block for each version? That would make the documentation easier to maintain as new versions are added - though on the other hand, if there are significant changes between versions, those would be more difficult to document all in a single block.

> That potentially becomes a performance consideration, because right now the SDL files on MOULa are downloaded in full on every launch of the game

That's a valid concern, but I think right now we don't have to worry about it too much. For shards that ship H'uru clients by default, it's generally not an issue, because the .sdl files can be (and usually are) provided via the file server, so they are downloaded in gzipped form and only redownloaded if they actually changed. It *would* be an issue for OpenUru clients, which normally redownload all .sdl files on every launch, but I assume the documentation comments won't be ported to the OpenUru repo, so they won't land on Minkata or in the deliveries to Cyan.

If large comments in .sdl files become a real practical problem at some point, I'm sure we can build a simple minification tool that automatically strips out comments and unneeded whitespace, which could be used to reduce the file size of .sdl files before they are deployed to a production shard.

**Comment by Deledrius** [comment]
It would be really great if Cyan could enable compression on the file transfers. :(  Oh well.

**Comment by Hoikas** [comment]
> Something to be aware of with SDL files is that every new change (adding or removing variables) requires copying and pasting the entire block of SDL declarations as a new version. If you're including inline comments in the file, those will probably also get copied/pasted each time.

Tangential, but something to consider here is that we have the SDL for many Ages since *before* GameTap and even *before ABM*. These older state descriptors could safely be deleted from the SDL files because it seems... highly unlikely that anyone has an SDL blob on their shard that from before the initial open source release. In short, we should consider deleting all state descriptors prior to the most recent versions from the [initial release](https://github.com/H-uru/moul-scripts/tree/db8ac4c0aa5467269c071010624df19c9878921d/SDL).

**Comment by Attagoat** [comment]
Age SDLs will be done first, I'll worry about non-age SDL's later

**Comment by TheScarFr** [comment]
> Perhaps it would make sense to document everything in a single block at the top of each .sdl file, rather than inside the block for each version?

Or, as the used version is always the one at the bottom of each .sdl file, maybe the documentation could be added after the last/current version?
